docs/topics/auth/passwords.txt

```diff
@@ -48,12 +48,17 @@ you're not sure, you probably don't.  If you do, please read on:
 
 Django chooses the algorithm to use by consulting the
 :setting:`PASSWORD_HASHERS` setting. This is a list of hashing algorithm
-classes that this Django installation supports. The first entry in this list
-(that is, ``settings.PASSWORD_HASHERS[0]``) will be used to store passwords,
-and all the other entries are valid hashers that can be used to check existing
-passwords.  This means that if you want to use a different algorithm, you'll
-need to modify :setting:`PASSWORD_HASHERS` to list your preferred algorithm
-first in the list.
+classes that this Django installation supports.
+
+For storing passwords, Django will use the first hasher in
+:setting:`PASSWORD_HASHERS`. To store new passwords with a different algorithm,
+modify :setting:`PASSWORD_HASHERS` to list your preferred algorithm first in
+the list.
+
+For verifying passwords, Django will look up the hasher from the list with
+the same algorithm as the stored password. If a stored password lists an
```

Comment:
Similar sort of note, so disregard as you choose. A stored password isn't the same _type_ of list. It does technically _list_ one in the saved form, but it might be better to say "includes" or "references" or "contains" maybe?

Reply:
Agree, changed to better reflect what's going on:

```
For verifying passwords, Django will look up the hasher from the list by the
algorithm name in the stored password.
```

Reply:
It's been long enough that, reading back what I wrote, I wasn't as clear as I'd hoped! A reminder that it's just a suggestion; this is the bit I was probably going for changing up more, emphasis mine:

> from the _list_ with the same algorithm as the stored password. If a stored password _lists_ an

It's the two types of "list" being referenced which I think is a bit on the stumbling side. I think my proposal was that you consider one of these:

>  If a stored password references an
>  If a stored password includes an
>  If a stored password contains an

Though you could also skirt the issue by doing something like:
> For verifying passwords, Django will find the hasher matching the algorithm name in the stored password.

Either/both probably work to mitigate the "list" overloading (damn flexible English!)

Reply:
Nice suggestions. I tried the verb "to name", just edited to:

```
For verifying passwords, Django will find the hasher in the list that matches
the algorithm name in the stored password. If a stored password names an
algorithm not found in :setting:`PASSWORD_HASHERS`, trying to verify it will
raise ``ValueError``.
```